portfolio/src/main/java/com/google/sps/servlets/UploadServlet.java

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.storage.Blob;
+import com.google.cloud.storage.BlobId;
+import com.google.cloud.storage.BlobInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import javax.servlet.ServletException;
+import javax.servlet.annotation.MultipartConfig;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+
+/**
+ * Takes an image submitted by the user and uploads it to Cloud Storage, and then displays it as
+ * HTML in the response.
+ */
+@WebServlet("/upload")
+@MultipartConfig
+public class UploadServlet extends HttpServlet {
+
+  @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response)
+        throws ServletException, IOException {
+
+            // Get the message entered by the user.
+            String message = request.getParameter("message");
+
+            // Get the file chosen by the user.
+            Part filePart = request.getPart("image");
+            String fileName = filePart.getSubmittedFileName();
+            InputStream fileInputStream = filePart.getInputStream();
+
+            // Upload the file and get its URL
+            String uploadedFileUrl = uploadToCloudStorage(fileName, fileInputStream);
+
+            // Output some HTML that shows the data the user entered.
+            // You could also store the uploadedFileUrl in Datastore instead.
+            PrintWriter out = response.getWriter();
+            out.println("<p>Here's the image you uploaded:</p>");
+            out.println("<a href=\"" + uploadedFileUrl + "\">");
+            out.println("<img src=\"" + uploadedFileUrl + "\" />");
+            out.println("</a>");
+            out.println("<p>Here's the text you entered:</p>");
+            out.println(message);
+        }
+
+    /** Uploads a file to Cloud Storage and returns the uploaded file's URL. */
+    private static String uploadToCloudStorage(String fileName, InputStream fileInputStream) {
+        String projectId = "rshillingford-sps-spring21";
```

Comment:
Same comment here as the previous file. If the variables will be non-changing, you can consider making them final static class variables. https://www.tutorialspoint.com/Final-static-variables-in-Java